js/modules/k6/experimental/csv/module.go

```diff
@@ -0,0 +1,289 @@
+// Package csv provides a CSV parser for k6.
+package csv
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"strconv"
+	"sync/atomic"
+	"time"
+
+	"go.k6.io/k6/js/modules/k6/data"
+
+	"github.com/grafana/sobek"
+
+	"gopkg.in/guregu/null.v3"
+
+	"go.k6.io/k6/js/promises"
+
+	"go.k6.io/k6/js/modules/k6/experimental/fs"
+
+	"go.k6.io/k6/js/common"
+	"go.k6.io/k6/js/modules"
+)
+
+type (
+	// RootModule is the global module instance that will create instances of our
+	// module for each VU.
+	RootModule struct {
+		// FIXME: we end up instantiating the data module ourselves, which would live side by side with other
```

Comment:
To be honest, I'm not really sure what's preferable in the tradeoff of global state vs having two instances of the data module 🤔

Reply:
I also don't see an issue in having two instances of data. But the current version relies on the global root's module state but initializes it in the module instance, the suggestion to move is relevant. Otherwise, we could just move the data to the instance :shrug:

Reply:
To make sure I don't misunderstand, the alternative would be to run the following (or equivalent) in the `RootModule` constructor?
```
	globalThis.data = require("k6/data");
```

Reply:
Okay, after re-checking the code, I think I missed that `Data` requires the VU for initialization. TBH, I felt that it was just the shared memory, then yeah, probably for a single initialization, the current way of initialization, even if it looks hacky, is the only option. So closing the tread :+1: